---
title: "Behavioral and ecological preferences drive z-distribution of rare species eDNA in pelagic marine environments"
format: html
editor: visual
execute:
  echo: false
  message: false
  warning: false
project:
  output-dir: docs/
  output-file: index.html
---

```{r}
library(tidyverse)
library(patchwork)

load("ProcessedData/detect_species_meta.RData")
load("ProcessedData/detectDepth_plots.Rdata")
load("ProcessedData/H1models.RData")

metadata <- read.csv("./Data/Hake_2019_metadata.csv")
```

## Introduction

All analyses of community composition face a similar challenge: rare species dropout,. This leads to innacurate estimates of biodiversity or community composition and greater difficulty assessing changes in ecosystem resilience, as well as innacurate characterization of a rare species distribution and greater difficulty assessing changes in distribution in response to environmental perterbations.

Many methodological approaches have been developed to address this problem, both in the experimental design phase (REFS) and in the data analysis phase (REFS). Statisical approaches aim to quantify and minimize the effect of rare species dropout, while sampling approaches aim to reduce the rate of dropout via robust sampling design. Within the latter category, it has been shown that adopting an ensemble approach, i.e. integrating across multiple types of observation methods in order to smooth across the weaknesses of each approach individually, often produces more robust estimates of species' distributions and community composition than using single methods alone.

eDNA is emerging as a viable tool for improving spatial distribution models for rare species, especially when used jointly with existing sampling methods, e.g. visual and passive acoustic surveys for marine mammals, or net tows, active acoustics, and/or bottom trawls for fishes and invertebrates. In the case of marine mammals - notably some of the rarest species in the marine environment - two survey techniques are traditionally used. Visual surveys are most common, in which observers aboard a ship will record detections of any marine mammals within 3km of the ship's transect, covering the survey area in a systematic pattern. These surveys are limited to the surface of the ocean, where most marine mammals spend very little of their time, and to daylight hours - and their effectiveness decreases in poor weather or sea state. Acoustic surveys collect recordings along similar systematic transects, and can often detect species up to XX kilometers away from the ship in any direction. They do have the same limitations at visual surveys, but only detect species that are acitvely vocalizing. eDNA surveys, where water samples are collected along similar transects, and often at systematic point locations along the transect do not share limitations related to depth, daylight, weather conditions, or species' vocal activity, but notably have a much smaller detection range than either of the two previous surveys - likely 10s of meters rather than multiple kilometers.

In open ocean, deepwater environments, our understanding of species' distributions may be affected by sampling depth, a fundamental aspect of experimetal design often overlooked in 3D evironments. This is expecially true for mobile species. In the pelagic marine environment, some rare species have depth distributions spanning 3km or greater, which complicates decision-making regarding where to collect point-samples with the aim of describing x-y distribution, either independently of or alongside z-distribution.

Because eDNA samples have a limited radius of detection, and because they are costly both to collect and to process, studies targeting a specific set of rare species may benefit from restricting their range of sampling depths, especially when x-y distribution is of a higher priority. However, there is currently a lack of published data providing guidance regarding the range of sampling depths that can be used to target certain species, taxonomic groups, or ecological groups of rare species.

Here, we aim improve our understanding of the depth distribution of rare species' eDNA in the water column and identify major environmental and ecological drivers of observed patterns. We specifically ask three questions:

1. Does eDNA distribution vary in the z-dimension? If so, what major environmental or ecological factors can be used to predict eDNA distribution across depths?
2. Does the effect of biological replication vary in the z-dimension? 
3. Does x-y variability in eDNA distribution affect variability in the z-dimension?

To answer these questions, we utilize eDNA samples collected throughout the California Current from the northern extent of the United States' EEZ to the latitude of XXX in southern California, and between XXX and XXX miles offshore. By collapsing eDNA detections across x and y dimensions and focusing only on the z-distribution of sampling effort (0-500 m), we aim to determine whether species- or taxon-specific detection probability varies with depth and how depth-dependent detection probability varies with increased sampling replication. Finally, we explicitly re-introduce x-y dimensions in order to determine whether depth-dependent detection probability varies with latitude or longitude.

## Methods

### Sample Collection

We collected water samples during biennial NOAA surveys quantifying hake abundance and distribution, in tandem with nightly CTD casts, as described in Shelton et al. \[-\@shelton_etal22\]. We targeted 0, 50, 150, 300, and 500 m depth strata along transects with a western terminus at the U.S. EEZ (300 km from shore) and an eastern terminus at the continental shelf break (generally \~XX km from shore).

### Sample Processing

We extracted DNA from all samples using a phenol-chloroform extraction protocol as described in XXX (Shaffer? Valdivia? another?). We then amplified each sample using three different primers: MiFish Universal (hereafter MFU, REF), targeting the 12S region of fishes, which often captures marine mammals; Marine Vertebrates 1 (hereafter MarVer1, REF), targeting the 12S region of marine vertebrates broadly; and D-loop (hereafter DL, REF) targeting the control region of cetaceans specifically. We amplified one technical replicate each for MFU and MV1 primers, and three technical replicates for D-loop. Our PCR amplification protocols

### Bioinformatic Processing

All raw sequence data were processed using a custom dada2-based pipeline and custom reference database based on NCBI sequence data. The pipeline and code for reference database generation can be found at \[GITHUB REPO HERE\]. Breifly, we ...

Once the raw sequences were trimmed, filtered, merged, clustered to ASVs, and once chimeras were removed, we assigned ASV taxonomy using ...

For each primer region, we built a custom reference database using all available sequence data from NCBI with the following filtering paramaeters. First, because the 12S mtDNA region has generally low resolution across many cetacean species, we filtered the reference database to exclude any species not observed in the North Pacific Ocean, excepting a few species known to occasionally visit from nearby regions, e.g. bowhead and beluga whales. We also exclude any pf NCBI's sequence IDs that were not manually validated by a human, due to the fact that NCBI's automated ID pipeline is still in beta mode and can produce erroneous identifications (REF here?).

### Data QAQC

To obtain detections, we defined a detection as any species/station/depth/technical replicate with at least one sequence read, and extracted all mammal detections across the three primer sets. Using this data set as a starting point, we filtered out any non-target (non-cetacean) species, and also removed any detections of bottlenose dolphins and unidentified Balaenopterids.

To obtain non-detections, we included all MFU and MV1 samples that had at least one marine species detected, but no marine mammals - excluding any samples that completely failed sequencing. Because our DL primer specifically targets a subset of very rare species, we assume that sequencing failure indicates a non-detection - therefore, for this primer we included any station/depth/species combination that was not detected as a non-detect.

Once all detections and non-detections were combined, we doncuted a final processing step to account for the fact that our DNA detections are not differentiated among living and dead animals. We therefore examined the dataset for potential whalefall (species detected outside their known depth range and within 100 m of the ocean bottom), potential killer whale prey (known prey speces detected outside their known depth range and at the same station as a killer whale), and potential shark prey (known prey speces detected outside their known depth range and at the same station as a shark).

### Data Analysis

## Results

We collected `r metadata %>% group_by(station, depth) %>% distinct() %>% nrow()` water samples during the 2019 NOAA hake abundance survey, target 0, 50, 150, 300, and 500 m sampling depths at each station. Where the bottom was less than 500 m deep, we did not collect a sample at 500 m. Of these water samples, we successfully sequenced `r length(unique(samples_info_onedil$NWFSCsampleID))` samples, including one replicate of `r samples_info_onedil %>% filter(primer == "MFU") %>% nrow()`MFU1 samples, `r samples_info_onedil %>% filter(primer == "MV1" & techRep == 1) %>% nrow()` MV1 samples with a second technical replicate for `r samples_info_onedil %>% filter(primer == "MV1" & techRep == 2) %>% nrow()`, and a third technical replicate for `r samples_info_onedil %>% filter(primer == "MV1" & techRep == 3) %>% nrow()` samples, three technical replicates for `r samples_info_onedil %>% filter(primer == "DL" & techRep == 3) %>% nrow()` DL samples and five technical replicates for `r samples_info_onedil %>% filter(primer == "DL" & techRep == 5) %>% nrow()` DL samples.

Across all samples, we collected `r sum(detect_species_meta$Detected)` detections of `r length(unique(detect_species_meta$BestTaxon))` cetcean species. Figure 1 illustrates the distribution of detections across sampling depths, ignoring x-y distribution, for all cetacean species included in the study.

```{r}
#| label: fig-bubble
#| fig-cap: "Distribution of detections across sampling depths for all species included in the study. Species are grouped as follows: Baleen whales includes the Balaenopteridae, Balaenidae, and Eschericthidae; Dolphins/Porpoises includes all Delphinidae and Phocoenidae; Beaked whales includes all Ziphiidae."
#| 
detectDepth_bubble
```

### Does detection probability vary with depth?

GAM-based model selection of detection probablity across sampling depths negated the null hypothesis of no variation across depth, and indicated that the best alternative model includes two covariates: sampling depth factored by species, and time-at-depth factored by species.

### Does the effect of technical replication on probability of detection vary with depth?

### Does depth variability in probability of detection vary across latitude or longitude?
## Discussion
